src/components/common/pills/pills.tsx

```diff
@@ -0,0 +1,145 @@
+import { PillItem } from "./pill-items";
+import { AddMultiEntryButton } from "./pills.styles";
+
+interface IProps {
+    id: string;
+}
+
+export const PillFields = ({ id }: IProps) => {
+    // =========================================================================
+    // CONST, STATE, REF
+    // =========================================================================
+    const {
+        control,
+        formState: { errors },
+    } = useFormContext<ITextareaFieldAttributes>();
+    const { fields, append, remove, move } = useFieldArray({
+        control,
+        name: "pillItems",
+        shouldUnregister: true,
+    });
+
+    const sensors = useSensors(
+        useSensor(PointerSensor, { activationConstraint: { distance: 8 } }),
+        useSensor(KeyboardSensor, {
+            coordinateGetter: sortableKeyboardCoordinates,
+        })
+    );
+
+    const disableDeleteButton = fields.length <= 2;
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDragEnd = (event: DragEndEvent) => {
+        const { active, over } = event;
+
+        if (over && active.id !== over.id) {
+            const oldIndex = fields.findIndex((item) => item.id === active.id);
+            const newIndex = fields.findIndex((item) => item.id === over.id);
+            move(oldIndex, newIndex);
+        }
+    };
+
+    const handleAddButtonClick = () => {
+        append({ content: "" });
+    };
+
+    const handleDeleteButtonClick = (index: number) => {
+        remove(index);
+    };
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        if (fields.length === 0) {
+            append([{ content: "" }, { content: "" }], {
+                shouldFocus: false,
+            });
+        }
+    }, []);
+
+    // =============================================================================
+    // RENDER FUNCTIONS
+    // =============================================================================
+    const renderPillItems = () =>
+        fields.map((item, index) => (
+            <PillItem
+                key={item.id}
+                item={item}
+                index={index}
+                onDelete={() => handleDeleteButtonClick(index)}
+                disableDelete={disableDeleteButton}
+            />
+        ));
+
+    return (
+        <DndContext
+            sensors={sensors}
+            collisionDetection={closestCenter}
+            onDragEnd={handleDragEnd}
+        >
+            <SortableContext
+                items={fields}
+                strategy={verticalListSortingStrategy}
+            >
+                <Form.Label>Pill Items</Form.Label>
+                <div>{fields.length > 0 && renderPillItems()}</div>
```

Comment:
- remove div as it has no use here
- length check is not necessary, if there is a need to against null fields, apply optional chaining in `renderPillItems()`

```suggestion
                {renderPillItems()}
```